ldi-core/ldes-client-v2/tree-node-supplier/src/main/java/ldes/client/treenodesupplier/repository/sqlite/SqliteMemberRepository.java

```diff
@@ -0,0 +1,48 @@
+package ldes.client.treenodesupplier.repository.sqlite;
+
+import ldes.client.treenodesupplier.domain.entities.MemberRecord;
+import ldes.client.treenodesupplier.domain.valueobject.MemberStatus;
+import ldes.client.treenodesupplier.repository.MemberRepository;
+
+import java.util.Optional;
+
+import javax.persistence.EntityManager;
+
+public class SqliteMemberRepository implements MemberRepository {
+	private final EntityManagerFactory entityManagerFactory = EntityManagerFactory.getInstance();
+	private final EntityManager entityManager = entityManagerFactory.getEntityManager();
+
+	@Override
+	public Optional<MemberRecord> getUnprocessedTreeMember() {
+
+		return entityManager
```

Comment:
Can you make this a typed query? It will remove the intellij warning and you will not need casting either:

		``return entityManager
				.createNamedQuery("Member.getByMemberStatus", MemberRecordEntity.class)
				.setParameter("memberStatus", MemberStatus.UNPROCESSED)
				.getResultStream()
				.map(MemberRecordEntity::toMemberRecord)
				.findFirst();``